src/NUnitFramework/framework/Internal/ExceptionHelper.cs

```diff
@@ -151,6 +166,24 @@ private static string GetExceptionMessage(Exception ex)
             return ex.Message;
         }
 
+        private static void AppendExceptionDataContents(Exception ex, StringBuilder sb)
+        {
+            if (ex.Data.Count == 0)
+            {
+                return;
+            }
+
+            sb.AppendLine();
+            sb.AppendLine("Data:");
+            foreach (DictionaryEntry kvp in ex.Data)
+            {
+                sb.Append("  ");
+                sb.Append(kvp.Key);
+                sb.Append(": ");
+                sb.AppendLine(kvp.Value?.ToString() ?? "<null>");
```

Comment:
I can change the appends.

Reply:
I added "Data:" because it's valuable to know why the values are being output.  I'm working on another PR to support registering delegates to output additional details.  Section headers will be very helpful.